types/src/validator_change.rs

```diff
@@ -19,32 +19,41 @@ impl<Sig: Signature> ValidatorChangeEventWithProof<Sig> {
             ledger_info_with_sigs,
         }
     }
+
+    pub fn epoch(&self) -> Option<u64> {
```

Comment:
Would be helpful to understand the intended semantics of this method given there is a vector of LedgerInfoWithSignatures stored.

This appears to return the first epoch of the epoch proof.  Maybe we can rename it to initial_epoch() or something better?

Reply:
The intended semantics is to return the lowest epoch of the proof, I was trying to unify every message to have an `epoch` function to indicate which epoch could process it